src/StateReaderTaskEither.ts

```diff
@@ -850,3 +850,47 @@ export const apS: <A, N extends string, S, R, E, B>(
 ) => (
   fa: StateReaderTaskEither<S, R, E, A>
 ) => StateReaderTaskEither<S, R, E, { [K in keyof A | N]: K extends keyof A ? A[K] : B }> = apSW
+
+// -------------------------------------------------------------------------------------
+// array utils
+// -------------------------------------------------------------------------------------
+
+/**
+ * @since 2.9.0
+ */
+export const traverseArrayWithIndex: <S, R, E, A, B>(
```

Comment:
This looks more like `traverseSeqArrayWithIndex`, right? I guess we also want the "parallel" version.

Reply:
I don't think it can be done, in State Monad you need the last state from the previous monad
and If you look at this file you can see that `stateReaderTaskEither` and `stateReaderTaskEitherSeq` are the same.